kubernetes/samples/scripts/domain-lifecycle/startServer.sh

```diff
@@ -0,0 +1,143 @@
+  resource to 'ALWAYS'. It also increases the 'spec.clusters[<cluster-name>].replicas'
+  value for the managed server's cluster by '1'.  The 'spec.clusters[<cluster-name>].replicas'
+  value can be kept constant by using '-k' option.
+ 
+  Usage:
+ 
+    $(basename $0) -s myserver [-n mynamespace] [-d mydomainuid] [-k] {-m kubecli]
+  
+    -s <server_name>           : Server name parameter is required.
+
+    -d <domain_uid>            : Domain unique-id. Default is 'sample-domain1'.
+
+    -n <namespace>             : Domain namespace. Default is 'sample-domain1-ns'.
+
+    -k <keep_replica_constant> : Keep replica count constant. Default behavior is to increment replica count.
+
+    -m <kubernetes_cli>        : Kubernetes command line interface. Default is 'kubectl'.
+
+    -h                         : This help.
+   
+EOF
+exit $1
+}
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+serverName=""
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+keepReplicaConstant=false
+
+while getopts "kd:n:m:s:h" opt; do
+  case $opt in
+    s) serverName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    k) keepReplicaConstant=true;
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  # Validate the required files exist
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  # Validate that server name parameter is specified.
+  if [ -z "${serverName}" ]; then
+    validationError "Please specify a server name using '-s' parameter e.g. '-s managed-server1'."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the domain in json format
+domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json)
+if [ $? -ne 0 ]; then
+  echo "[ERROR} Unable to get domain resource. Please make sure 'domain_uid' and 'namespace' provided with '-d' and '-n' arguments are correct."
+  exit 1
+fi
+
+# check if server pod is already running
+${kubernetesCli} get pod ${domainUid}-${serverName} -n ${domainNamespace} > /dev/null 2>&1
+if [ $? -eq 0 ]; then
+  echo "Server pod ${domainUid}-${serverName} already exists. Exiting."
+  exit 1
+fi
+
+# Validate that specified server is either part of a cluster or is an independent managed server
+validateServerAndFindCluster "${domainUid}" "${domainNamespace}" isValidServer clusterName
+if [ "${isValidServer}" != 'true' ]; then
+  printError "Server ${serverName} is not part of any cluster and it's not an independent managed server. Please make sure that server name specified is correct."
+  exit 1
+fi
+
+# Create server start policy patch with ALWAYS value
+currentPolicy=""
+serverStartPolicy=ALWAYS
+createServerStartPolicyPatch "${domainJson}" "${serverName}" "${serverStartPolicy}" serverStartPolicyPatch currentPolicy
```

Comment:
I think the check to see if the server pod is up ^^^ might be misleading - the existence/non-existence of a server pod does not always correctly reflect it's current server-start-policy.

Reply:
Ok, Originally, I had thought that the goal of this script was to start a  server pod and I wanted to exit early when server pod already existed. That was the reason for above check. I will change so that the server start policy is patched even when pod exists.

Reply:
Let's discuss this tomorrow. I think I'm missing some context and I don't want to add behavior to the script the is surprising or nonintuitive.

Reply:
Thanks Ryan. I have added an extra check to see if the server start policy is ALWAYS and exit early. It would be good if we can discuss the behavior tomorrow to make sure if it's not surprising or nonintuitive.